### PR TITLE
"Alert" component - Showcase

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

In this PR I have updated the showcase for the `Alert` component so that:
- it's ready for all the other missing types of alert
- it's more compact and easy to visually scan

_Notice: to avoid conflicts and facilitate the code review, I have added a new section for the showcase. Once approved this, I'll delete the old one._

### :camera_flash: Screenshots

localhost_3000_components_alert

***

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [x] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by amyrlam — APPROVED** [review]
:+1:

Ignoring the other non-Alert Percy's since not going into main

**Comment by didoo** [comment]
@amyrlam I'll merge this once the other component-related ones are merged too, because there will be conflicts and will require some small update. I'll ask you a second pass then. In the meantime thanks for the first review.

**Comment by amyrlam** [comment]
@didoo Updated https://github.com/hashicorp/design-system/pull/91 so can see the new icons in Alert (Compact) now

To the question I asked aloud on the other PR, package.json updates will come programmatically when needed, like this https://github.com/hashicorp/design-system/pull/102/files

It's kind of confusing, that some things are out of sync in the package.json like https://github.com/hashicorp/design-system/blob/main/packages/ember-flight-icons/package.json#L38 not being 2.2.0 https://www.npmjs.com/package/@hashicorp/flight-icons, but I think we just live with that in Changesets world from what I've read cc @Dhaulagiri 

I'm signing off for the day, short day, hope to be back to normalcy tmrw

**Pull request merged**
